### PR TITLE
msx.cpp: Cartridge slot soundin and clock pins.

### PR DESCRIPTION
**Pull request by wilbertpol**
Add device_mixer_interface to the cartridge and let cartridges output their sound through this.
Set the clock on the cartridge slot and let cartridges derive their clock from it.

### CODE REVIEW

File: src/devices/bus/msx/cart/konami.cpp [hunk 1]
```diff
@@ -272,9 +272,9 @@ msx_cart_synthesizer_device::msx_cart_synthesizer_device(const machine_config &m
 
 void msx_cart_synthesizer_device::device_add_mconfig(machine_config &config)
 {
-	// This is actually incorrect. The sound output is passed back into the MSX machine where it is mixed internally and output through the system 'speaker'.
-	SPEAKER(config, "speaker").front_center();
-	DAC_8BIT_R2R(config, m_dac, 0).add_route(ALL_OUTPUTS, "speaker", 0.3); // unknown DAC
+	DAC_8BIT_R2R(config, m_dac, 0);
+	if (parent_slot())
+		m_dac->add_route(ALL_OUTPUTS, soundin(), 0.3, AUTO_ALLOC_INPUT, 0); // unknown DAC
```
Comment (cuavas) [inline src/devices/bus/msx/cart/konami.cpp:277]: The “unknown DAC” comment should go on the line with the DAC itself.

File: src/devices/bus/msx/cart/majutsushi.cpp [hunk 2]
```diff
@@ -19,9 +19,9 @@ msx_cart_majutsushi_device::msx_cart_majutsushi_device(const machine_config &mco
 
 void msx_cart_majutsushi_device::device_add_mconfig(machine_config &config)
 {
-	// This is actually incorrect. The sound output is passed back into the MSX machine where it is mixed internally and output through the system 'speaker'.
-	SPEAKER(config, "speaker").front_center();
-	DAC_8BIT_R2R(config, m_dac, 0).add_route(ALL_OUTPUTS, "speaker", 0.05); // unknown DAC
+	DAC_8BIT_R2R(config, m_dac, 0);
+	if (parent_slot())
+		m_dac->add_route(ALL_OUTPUTS, soundin(), 0.15, AUTO_ALLOC_INPUT, 0); // unknown DAC
```
Comment (cuavas) [inline src/devices/bus/msx/cart/majutsushi.cpp:24]: The “unknown DAC” comment should go on the line with the DAC itself.

**Review by cuavas — COMMENTED** [review]
Should the one-to-four slot expanders also be routing sound from their subslots back up to their parent slot, or do they not connect the sound input pin to the subslots?

File: src/devices/bus/msx/cart/majutsushi.cpp (hunk 2 above)
Comment (wilbertpol) [reply src/devices/bus/msx/cart/majutsushi.cpp]: moved the comment

File: src/devices/bus/msx/cart/konami.cpp (hunk 1 above)
Comment (wilbertpol) [reply src/devices/bus/msx/cart/konami.cpp]: moved the comment

**Comment by wilbertpol** [comment]
> Should the one-to-four slot expanders also be routing sound from their subslots back up to their parent slot, or do they not connect the sound input pin to the subslots?

Yes, they should, and also the clock should be propagated to the subslots. I had totally forgotten about that. Thanks and fixed.

**Pull request merged**